### PR TITLE
Accept specific output name for `focus-output` and `send-to-ouput` commands

### PR DESCRIPTION
**Pull request by pmkap**
This allows to focus a specific output or send the view to a specific output.

With riverctl it looks like this:
`send-to-output next|previous|up|right|down|left|name`
`focus-output next|previous|up|right|down|left|name`

After matching the argument for next/previous and up/right/etc., the output names itself are checked for a match.

Exmaples:
`riverctl focus-output WL-1`
`riverctl send-to-output HDMI-A-1`

### CODE REVIEW

**Comment by nwg-piotr** [comment]
Works excellent, and should allow to write a script to workaround #448.

File: doc/riverctl.1.scd [hunk 1]
```diff
@@ -53,6 +53,9 @@ over the Wayland protocol.
 *focus-output* *next*|*previous*|*up*|*right*|*down*|*left*
 	Focus the next or previous output or the closest output in any direction.
 
+*focus-output-name* _output_
+	Focus the a specific output by its name.
```
Comment (and3rson) [inline doc/riverctl.1.scd:57]: Seems like a typo: "the a"

**Comment by Grawp** [comment]
@pmkap Would you consider updating the PR so it builds with Zig 0.9.0? Thanks.
Btw. the PR itself is very helpful. I had been using it as patch for River builds in NixOS till 0.9.0 broke it.

**Comment by pmkap** [comment]
@Grawp Sure, I'll update this somewhen in the next days.

Also, I'm thinking maybe it would be nicer not to have an extra command but just something like
`focus-output next|previous|output-name` and resp.
`send-to-output next|previous|output-name`

Or is there anything that speaks against this?

File: doc/riverctl.1.scd [hunk 2]
```diff
@@ -50,8 +50,9 @@ over the Wayland protocol.
 	Remove an app-id or title from the float filter list. Note that this
 	affects only new views, not already existing ones.
 
-*focus-output* *next*|*previous*|*up*|*right*|*down*|*left*
-	Focus the next or previous output or the closest output in any direction.
+*focus-output* *next*|*previous*|*up*|*right*|*down*|*left*|*name*
```
Comment (Leon-Plickat) [inline doc/riverctl.1.scd:53]: Should be `_name_`.

File: doc/riverctl.1.scd [hunk 3]
```diff
@@ -68,9 +69,9 @@ over the Wayland protocol.
 	Snap the focused view to the specified screen edge. The view will
 	be set to floating.
 
-*send-to-output* *next*|*previous*|*up*|*right*|*down*|*left*
-	Send the focused view to the next or previous output or the closest
-	output in any direction.
+*send-to-output* *next*|*previous*|*up*|*right*|*down*|*left*|*name*
```
Comment (Leon-Plickat) [inline doc/riverctl.1.scd:72]: same here.

**Review by Leon-Plickat — CHANGES_REQUESTED** [review]

File: doc/riverctl.1.scd (hunk 2 above)
Comment (pmkap) [reply doc/riverctl.1.scd:53]: Ah yeah I can see why.

**Review by Leon-Plickat — APPROVED** [review]

**Comment by pmkap** [comment]
Currently, when having only one output and using e.g `riverctl send-to-output next`, the view gets pushed on top resp. bottom of the stack. Also with this proposed PR the user can send the view to the current output. I wonder if this is a bit unintuitive and we should intercept this case: `seat.focused.view.output == destination_output`.

**Comment by ifreund** [comment]
> Currently, when having only one output and using e.g `riverctl send-to-output next`, the view gets pushed on top resp. bottom of the stack. Also with this proposed PR the user can send the view to the current output. I wonder if this is a bit unintuitive and we should intercept this case: `seat.focused.view.output == destination_output`.

Yeah, this should do absolutely nothing when targeting the current output.

File: river/command/output.zig [hunk 4]
```diff
@@ -93,6 +95,13 @@ fn getOutput(seat: *Seat, str: []const u8) !?*Output {
         ) orelse return null;
         return @intToPtr(*Output, wlr_output.data);
     } else {
+        // Check if an output matches by name
+        var it = server.root.outputs.first;
+        while (it) |node| : (it = node.next) {
+            if (std.mem.eql(u8, std.mem.sliceTo(node.data.wlr_output.name, 0), str)) {
```
Comment (ifreund) [inline river/command/output.zig:101]: We actually don't need `sliceTo()` here as `wlr_output.name` is now a 0 terminated pointer not an array.
```suggestion
            if (std.mem.eql(u8, std.mem.span(node.data.wlr_output.name), str)) {
```

Comment (ifreund) [reply river/command/output.zig:101]: And looks like I forgot to change the doc comment in zig-wlroots...

Comment (pmkap) [reply river/command/output.zig:101]: Ah nice!

**Review by ifreund — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**